webui/react/src/pages/ProjectDetails.tsx

```diff
@@ -757,26 +757,6 @@ const ProjectDetails: React.FC = () => {
     }
   }, [ total, settings.tableOffset, settings.tableLimit, updateSettings ]);
 
-  /*
-   * Get new experiments based on changes to the
-   * filters, pagination, search and sorter.
-   */
-  useEffect(() => {
-    fetchExperiments();
-    setIsLoading(true);
-  }, [
-    fetchExperiments,
-    settings.archived,
-    settings.label,
-    settings.search,
-    settings.sortDesc,
-    settings.sortKey,
-    settings.state,
-    settings.tableLimit,
-    settings.tableOffset,
-    settings.user,
-  ]);
-
   useEffect(() => {
     return () => canceler.abort();
```

Comment:
I don't know this for sure, but this suspiciously seems like API calls are being made continuously but previous calls are not cancelled properly and allowed to linger until completion. So with browser HTTP calls, there is a rough limit of about 6-8 per host and once it hits that limit it puts all subsequent calls in Pending. So I **think** the problem is that the canceler needs to be created for each call appropriately.

Reply:
That's a great perspective!